File: tests/const/const.cpp
#include <iostream>
#include <string>

int main() {
    const int max = 100;
    const int min = -5;
    const float pi = 3.14;
    const std::string greeting = "Hello, World!";
    const bool flag = true;
    
    int i;
    float r;
    
    i = 0;
    while ((i < max)) {
        i = (i + 1);
    }
    std::cout << i << std::endl;
    r = (pi * 2.0);
    std::cout << r << std::endl;
    std::cout << greeting << std::endl;
    i = (min + 10);
    std::cout << i << std::endl;
    if (flag) {
        std::cout << "flag is true" << std::endl;
    }
    else {
        std::cout << "flag is false" << std::endl;
    }
    return 0;
}

File: tests/const/const.pas
program ConstTest;

const
  MAX = 100;
  MIN = -5;
  PI = 3.14;
  GREETING = 'Hello, World!';
  FLAG = true;

var
  i: integer;
  r: real;

begin
  { использование целочисленной константы в цикле }
  i := 0;
  while i < MAX do
  begin
    i := i + 1
  end;
  writeln(i);

  { арифметика с константами }
  r := PI * 2.0;
  writeln(r);

  { строковая константа }
  writeln(GREETING);

  { отрицательная константа }
  i := MIN + 10;
  writeln(i);

  { boolean константа }
  if FLAG then
    writeln('flag is true')
  else
    writeln('flag is false')
end.
